src/Renderer/Camera.js

```diff
@@ -100,4 +101,26 @@ Camera.prototype.box3SizeOnScreen = function box3SizeOnScreen(box3, matrixWorld)
     return tmp.box3;
 };
 
+ /**
+ * Test for collision between camera and a geometry layer (DTM/DSM) to adjust camera position
+ * It could be modified later to handle an array of geometry layers
+ * @param {view} view where we test the collision between geometry layers and the camera
+ * @param {elevationLayer} elevationLayer (DTM/DSM) used to test the collision with the camera. Could be another geometry layer
+ * @param {minDistanceCollision} minDistanceCollision the minimum distance allowed between the camera and the surface
+ */
+Camera.prototype.adjustAltitudeToAvoidCollisionWithLayer = function adjustAltitudeToAvoidCollisionWithLayer(view, elevationLayer, minDistanceCollision) {
+    const camLocation = new Coordinates(view.referenceCrs, view.camera.camera3D.position).as('EPSG:4326');
+    if (elevationLayer !== undefined) {
```

Comment:
IMHO, either you:
- test for `undefined` and throw a useful error
- document that `elevationLayer` cannot be undefined

But testing and then ignoring `undefined` seems like a bad idea.

Reply:
I understand your comment but here, it's not an error. If undefined it means there is no collision it's not a proper error

Reply:
IMHO it makes no sense to call this function without an elevationLayer.
So cluttering the code with unneeded `if X !== undefined` isn't bringing us any value.

So I'd prefer this function to have a clear contract ("I require an elevationLayer") and leave it to the user the task of not calling it with an invalid value.